### DISCUSSION TITLE
"Recipes" for implementing specific module/library options by Gradle/Chaquopy?

### DISCUSSION THREAD

**Discussion by MarkusPiotrowski** [beeware/beeware#207] [Q&A] [answered]
This question is related to issue #181. I did some work to get some simple Bluetooth LE functionality in a BeeWare/Toga app on Android. My idea is now to provide a small library which can be used as replacement for bleak (an otherwise superb Python cross-platform library for using BLE) in BeeWare projects on Android.
To make this running:
 - permissions must be declared in AndroidManifest.xml
 - static proxies must be declared in build.gradle

My question is: Is there a way to provide this information so that it will be 'automatically' imported and set when developing an Android app with BeeWare/Chaquopy?

**Comment by mhsmith** [comment]
build.gradle settings can be added to an app using [`build_gradle_extra_content`](https://briefcase.readthedocs.io/en/stable/reference/platforms/android.html#build-gradle-extra-content) in pyproject.toml. There's not currently any equivalent for AndroidManifest.xml, though there's a suggestion to add it at https://github.com/beeware/briefcase/issues/547#issuecomment-1324930536.

**Comment by MarkusPiotrowski** [comment]
OK, suggest I have a (pure Python) library hosted on GitHub, which includes a `pyproject.toml` file with the section `build_gradle_extra_content`. 
In my BeeWare app, I also have a `pyproject.toml` file which, in its '[tool.briefcase.app.myapp.android]' section, links to that library as required.
Would briefcase automatically take over the `build_gradle_extra_content` into the app's `pyproject.toml` file or even into the `build.gradle` file?

**Reply by mhsmith** [reply, marked as answer]
No, each app would need to add the setting individually. To make it automatic, we would have to define a way of storing it in the library's metadata (because the pyproject.toml file isn't included in a wheel), and modify Briefcase to look for that.

For previous discussion, see https://github.com/beeware/rubicon-java/discussions/75#discussioncomment-3310423.

**Reply by MarkusPiotrowski** [reply]
Thank you.
